docs/reference/search/aggregations/reducers/movavg-reducer.asciidoc

```diff
@@ -0,0 +1,296 @@
+[[search-aggregations-reducers-movavg-reducer]]
+=== Moving Average Aggregation
+
+Given an ordered series of data, the Moving Average aggregation will slide a window across the data and emit the average
+value of that window.  For example, given the data `[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]`, we can calculate a simple moving
+average with windows size of `5` as follows:
+
+- (1 + 2 + 3 + 4 + 5) / 5  = 3
+- (2 + 3 + 4 + 5 + 6) / 5  = 4
+- (3 + 4 + 5 + 6 + 7) / 5 = 5
+- etc
+
+Moving averages are a simple method to smooth sequential data.  Moving averages are typically applied to time-based data,
+such as stock prices or server metrics.  The smoothing can be used to eliminate high frequency fluctuations or random noise,
+which allows the lower frequency trends to be more easily visualized, such as seasonality.
+
+==== Syntax
+
+A `moving_avg` aggregation looks like this in isolation:
+
+[source,js]
+--------------------------------------------------
+{
+    "movavg": {
+        "buckets_path": "the_sum",
+        "model": "double_exp",
+        "window": 5,
+        "gap_policy": "insert_zero",
+        "settings": {
+            "alpha": 0.8
+        }
+    }
+}
+--------------------------------------------------
+
+.`moving_avg` Parameters
+|===
+|Parameter Name |Description |Required |Default
+
+|`buckets_path` |The path to the metric that we wish to calculate a moving average for |Required |
+|`model` |The moving average weighting model that we wish to use |Optional |`simple`
+|`gap_policy` |Determines what should happen when a gap in the data is encountered. |Optional |`insert_zero`
+|`window` |The size of window to "slide" across the histogram. |Optional |`5`
+|`settings` |Model-specific settings, contents which differ depending on the model specified. |Optional |
+|===
+
+
+`moving_avg` aggregations must be embedded inside of a `histogram` or `date_histogram` aggregation.  They can be
+embedded like any other metric aggregation:
+
+[source,js]
+--------------------------------------------------
+{
+    "my_date_histo":{                <1>
+        "date_histogram":{
+            "field":"timestamp",
+            "interval":"day",
+            "min_doc_count": 0      <2>
+        },
+        "aggs":{
+            "the_sum":{
+                "sum":{ "field": "lemmings" } <3>
+            },
+            "the_movavg":{
+                "moving_avg":{ "buckets_path": "the_sum" } <4>
+            }
+        }
+    }
+}
+--------------------------------------------------
+<1> A `date_histogram` named "my_date_histo" is constructed on the "timestamp" field, with one-day intervals
+<2> We must specify "min_doc_count: 0" in our date histogram that all buckets are returned, even if they are empty.
```

Comment:
I dont't like it much in terms of usability. I think we really need to make min_doc_count=0 the default on histograms